pkg/plugins/runtime/k8s/metadata/annotations.go

```diff
@@ -118,6 +118,9 @@ const (
 
 	// KumaServiceName points to the Service that a MeshService is derived from
 	KumaServiceName = "k8s.kuma.io/service-name"
+
+	// HeadlessService is "true" when the Service had ClusterIP: None, otherwise "false"
+	HeadlessService = "k8s.kuma.io/headless-service"
```

Comment:
so if we include this to every Kube MeshService, how about
`k8s.kuma.io/service-type: headless|clusterip(or regular?)`. I think it's more future proof if we need another value